context.go

```diff
@@ -108,17 +108,27 @@ func (c *Context) getLoggerModule(name string, tags []string) *module {
 		// First tag wins when setting the logger tag from the config tag
 		// level cache. If there are no tag configs, then fallback to
 		// UNSPECIFIED and inherit the level correctly.
-		if configLevel, ok := c.modulesLabelConfig[tag]; ok && level == UNSPECIFIED {
+		if configLevel, ok := c.modulesTagConfig[tag]; ok && level == UNSPECIFIED {
 			level = configLevel
 		}
 	}
+
+	// As it's not possible to modify the parent's labels, it's safe to copy
+	// them at the time of creation. Otherwise we have to walk the parent chain
+	// to get the full set of labels for every log message.
+	labels := make(Labels)
+	for k, v := range parent.labels {
+		labels[k] = v
+	}
+
 	impl = &module{
-		name:         name,
-		level:        level,
-		parent:       parent,
-		context:      c,
-		tags:         tags,
-		labelsLookup: labelMap,
+		name:       name,
+		level:      level,
+		parent:     parent,
+		context:    c,
+		tags:       tags,
+		tagsLookup: labelMap,
```

Comment:
`labelMap` should probably also be `tagMap`

Reply:
Ah, right, I'll follow up.